R/dal.R

```diff
@@ -458,7 +459,65 @@ get_all_polio_data <- function(
                                    dplyr::pull(file))
       cli::cli_process_done()
 
-      cli::cli_process_start("13) Clearing out unused memory")
+      cli::cli_process_start("13) Creating Metadata object")
+
+      polis.cache <- edav_io(io = "read",
+                             file_loc = dplyr::filter(dl_table, grepl("cache", file)) |>
+                               dplyr::pull(file)) |>
+        dplyr::mutate(last_sync = as.Date(last_sync))
+
+      raw.data$metadata$download_time <- max(polis.cache$last_sync, na.rm = TRUE)
+
+      raw.data$metadata$processed_time <- edav_io(io = "list", file_loc = file.path(folder, "/polis")) |>
+        dplyr::filter(grepl("positives", name)) |>
+        dplyr::select("ctime" = "lastModified") |>
+        dplyr::mutate(ctime = as.Date(ctime)) |>
+        dplyr::pull(ctime)
+
+      raw.data$metadata$user <- polis.cache |>
+        dplyr::filter(table == "virus") |>
+        dplyr::pull(last_user)
+
+      raw.data$metadata$most_recent_pos <- max(raw.data[["pos"]]$dateonset, na.rm = TRUE)
```

Comment:
Recommend converting all list search operators to be atomic, ie: raw.data$pos$dateonset